src/SprykerEco/Zed/AvalaraTax/Business/Calculator/AbstractCartItemAvalaraTaxCalculator.php

```diff
@@ -0,0 +1,154 @@
+use Generated\Shared\Transfer\AvalaraCreateTransactionResponseTransfer;
+use Generated\Shared\Transfer\AvalaraTransactionLineTransfer;
+use Generated\Shared\Transfer\CalculableObjectTransfer;
+use Generated\Shared\Transfer\ItemTransfer;
+use SprykerEco\Zed\AvalaraTax\AvalaraTaxConfig;
+use SprykerEco\Zed\AvalaraTax\Business\Executor\AvalaraTransactionExecutorInterface;
+use SprykerEco\Zed\AvalaraTax\Dependency\Facade\AvalaraTaxToMoneyFacadeInterface;
+use SprykerEco\Zed\AvalaraTax\Dependency\Service\AvalaraTaxToUtilEncodingServiceInterface;
+
+abstract class AbstractCartItemAvalaraTaxCalculator implements CartItemAvalaraTaxCalculatorInterface
+{
+    /**
+     * @var \SprykerEco\Zed\AvalaraTax\Business\Executor\AvalaraTransactionExecutorInterface
+     */
+    protected $avalaraTransactionExecutor;
+
+    /**
+     * @var \SprykerEco\Zed\AvalaraTax\Dependency\Facade\AvalaraTaxToMoneyFacadeInterface
+     */
+    protected $moneyFacade;
+
+    /**
+     * @var \SprykerEco\Zed\AvalaraTax\Dependency\Service\AvalaraTaxToUtilEncodingServiceInterface
+     */
+    protected $utilEncodingService;
+
+    /**
+     * @var \SprykerEco\Zed\AvalaraTaxExtension\Dependency\Plugin\CreateTransactionRequestAfterPluginInterface[]
+     */
+    protected $createTransactionRequestAfterPlugins;
+
+    /**
+     * @param \SprykerEco\Zed\AvalaraTax\Business\Executor\AvalaraTransactionExecutorInterface $avalaraTransactionExecutor
+     * @param \SprykerEco\Zed\AvalaraTax\Dependency\Facade\AvalaraTaxToMoneyFacadeInterface $moneyFacade
+     * @param \SprykerEco\Zed\AvalaraTax\Dependency\Service\AvalaraTaxToUtilEncodingServiceInterface $utilEncodingService
+     * @param \SprykerEco\Zed\AvalaraTaxExtension\Dependency\Plugin\CreateTransactionRequestAfterPluginInterface[] $createTransactionRequestAfterPlugins
+     */
+    public function __construct(
+        AvalaraTransactionExecutorInterface $avalaraTransactionExecutor,
+        AvalaraTaxToMoneyFacadeInterface $moneyFacade,
+        AvalaraTaxToUtilEncodingServiceInterface $utilEncodingService,
+        array $createTransactionRequestAfterPlugins
+    ) {
+        $this->avalaraTransactionExecutor = $avalaraTransactionExecutor;
+        $this->moneyFacade = $moneyFacade;
+        $this->utilEncodingService = $utilEncodingService;
+        $this->createTransactionRequestAfterPlugins = $createTransactionRequestAfterPlugins;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CalculableObjectTransfer $calculableObjectTransfer
+     *
+     * @return void
+     */
+    public function calculateTax(CalculableObjectTransfer $calculableObjectTransfer): void
+    {
+        if (!$this->hasShipmentAddress($calculableObjectTransfer)) {
+            return;
+        }
+
+        $avalaraCreateTransactionResponseTransfer = $this->avalaraTransactionExecutor->executeAvalaraCreateTransaction(
+            $calculableObjectTransfer,
+            (string)AvalaraTaxConfig::AVALARA_TRANSACTION_TYPE_ID_SALES_ORDER
```

Comment:
We have to create two configuration methods:
* One would be used when we do temporary calculations before the order is created.
* Another one would be used when we do a calculation for a created order.

Reply:
Resolved.